### PR TITLE
[one-cmds] Support for FoldShape pass

### PR DESCRIPTION
**Pull request by miusic**
This commit supports FoldShape pass in one-cmds

---
Related to: #12046
Draft: #12407

ONE-DCO-1.0-Signed-off-by: jihunnn-kim <jihunnn.kim@samsung.com>

### CODE REVIEW

**Comment by miusic** [comment]
`./build/compiler/circle2circle/circle2circle --fold_shape ./input_model.circle ./output_model.circle`

---
with `input_model.circle`

input_model

`output_model.circle`

output_model

[input_and_output_circle_model.zip](https://github.com/Samsung/ONE/files/14109031/input_and_output_circle_model.zip)

**Comment by jinevening** [comment]
`onecc optimize --fold_shape ~~` would not work.

You have to add option to OPTIMIZATION_OPTS (not just O1).

File: compiler/one-cmds/onelib/constant.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ class CONSTANT:
         'fold_dwconv',
         'fold_fully_connected',
         'fold_gather',
+        'fold_shape',
```
Comment (zetwhite) [inline compiler/one-cmds/onelib/constant.py:32]: Isn't it necessary to add this option into `OPTIMIZATION_OPTION`(line 83) either?

Comment (zetwhite) [reply compiler/one-cmds/onelib/constant.py:32]: (oops, duplicated)

**Comment by miusic** [comment]
> You have to add option to OPTIMIZATION_OPTS (not just O1).

Updated!

**Review by zetwhite — APPROVED** [review]
LGTM

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**